tests/rpm_tests/create_rpm_test.py

```diff
@@ -0,0 +1,389 @@
+                ModifyingFilters.apply_dbus_prefix,
+                ModifyingFilters.apply_share_anaconda_prefix
+            ], src_files
+        )
+
+        src_files = self._check_and_return_as_list(src_files)
+
+        self._check_files_in_rpm(src_files, rpm_files)
+
+    def test_dbus_service_files(self):
+        rpm_files = self._get_core_rpm_content()
+
+        rpm_files = filter(FileFilters.rpm_dbus_services_only, rpm_files)
+
+        rpm_files = self._check_and_return_as_list(rpm_files)
+
+        src_files = self._get_source_files()
+        src_files = filter(FileFilters.src_dbus_only, src_files)
+        src_files = filter(FileFilters.services_only, src_files)
+
+        src_files = self._apply_maps(
+            [
+                ModifyingFilters.remove_data_dbus_prefix,
+                ModifyingFilters.apply_services_prefix,
+                ModifyingFilters.apply_dbus_prefix,
+                ModifyingFilters.apply_share_anaconda_prefix
+            ], src_files
+        )
+
+        src_files = self._check_and_return_as_list(src_files)
+
+        self._check_files_in_rpm(src_files, rpm_files)
+
+    def test_anaconda_service_files(self):
+        rpm_files = self._get_core_rpm_content()
+
+        rpm_files = filter(FileFilters.rpm_systemd_only, rpm_files)
+
+        rpm_files = self._check_and_return_as_list(rpm_files)
+
+        src_files = self._apply_filters(
+            [
+                FileFilters.src_systemd_only,
+                FileFilters.makefiles_exclude,
+                lambda f: FileFilters.specific_file_exclude(self.ANACONDA_GENERATOR, f)
+            ], self._get_source_files()
+        )
+
+        src_files = self._apply_maps(
+            [
+                ModifyingFilters.remove_data_systemd_prefix,
+                lambda x: ModifyingFilters.apply_rpm_prefix("/usr/lib/systemd/system", x)
+            ], src_files
+        )
+
+        src_files = self._check_and_return_as_list(src_files)
+
+        self._check_files_in_rpm(src_files, rpm_files)
+
+    def test_anaconda_service_generator_file(self):
+        rpm_files = self._get_core_rpm_content()
+
+        rpm_files = filter(FileFilters.rpm_systemd_only, rpm_files)
+
+        rpm_files = self._check_and_return_as_list(rpm_files)
+
+        src_files = self._apply_filters(
+            [
+                FileFilters.src_systemd_only,
+                FileFilters.makefiles_exclude,
+                lambda f: FileFilters.specific_file_only(self.ANACONDA_GENERATOR, f),
+            ], self._get_source_files()
+        )
+
+        src_files = self._apply_maps(
+            [
+                ModifyingFilters.remove_data_systemd_prefix,
+                lambda x: ModifyingFilters.apply_rpm_prefix("/usr/lib/systemd/system-generators",
+                                                            x),
+            ], src_files
+        )
+
+        src_files = self._check_and_return_as_list(src_files)
```

Comment:
You can probably call `_check_and_return_as_list` for `src_files` and `rpm_files` in the `_check_files_in_rpm` method.

Reply:
I don't think it is a good idea. It is then confusing that you see `src_files` checked but nothing for `rpm_files`.

If something like that then I'll rather create one check for both `src_files` and `rpm_files`.